presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctions.java

```diff
@@ -148,29 +154,102 @@ private static long epochDaysInZone(TimeZoneKey timeZoneKey, long instant)
     public void testLocalTime()
             throws Exception
     {
-        long millis = new LocalTime(session.getStartTime(), DateTimeZone.UTC).getMillisOfDay();
-        functionAssertions.assertFunction("LOCALTIME", TimeType.TIME, toTime(millis));
+        Session localSession = testSessionBuilder()
+                .setSystemProperty("legacy_timestamp", "false")
+                .setTimeZoneKey(TIME_ZONE_KEY)
+                .setStartTime(new DateTime(2017, 3, 1, 10, 0, 0, 0, DateTimeZone.UTC).getMillis())
+                .build();
+        FunctionAssertions localAssertion = new FunctionAssertions(localSession);
+        localAssertion.assertFunctionString("LOCALTIME", TimeType.TIME, "15:45:00.000");
     }
 
     @Test
     public void testCurrentTime()
             throws Exception
     {
-        long millis = new LocalTime(session.getStartTime(), DateTimeZone.UTC).getMillisOfDay();
-        functionAssertions.assertFunction("CURRENT_TIME", TIME_WITH_TIME_ZONE, new SqlTimeWithTimeZone(millis, session.getTimeZoneKey()));
+        Session localSession = testSessionBuilder()
+                .setSystemProperty("legacy_timestamp", "false")
+                .setTimeZoneKey(TIME_ZONE_KEY)
+                .setStartTime(new DateTime(2017, 3, 1, 10, 0, 0, 0, DateTimeZone.UTC).getMillis())
```

Comment:
Add a test that is near midnight UTD:

```
.setStartTime(new DateTime(2017, 3, 1, 23, 50, 0, 0, DateTimeZone.UTC).getMillis())
;

// This check currently passes, showing once again that `assertFunctionString` should begone:
localAssertion.assertFunctionString("CURRENT_TIME", TIME_WITH_TIME_ZONE, "05:35:00.000 Asia/Kathmandu");

// this check currently fails, but obviously should pass too
localAssertion.assertFunction("CURRENT_TIME = TIME '05:35:00.000 Asia/Kathmandu'", BOOLEAN, true);
```

To prevent situations like that, consider adding `check millis in [0, 86400,000)` in `SqlTime[WithTimeZone]` constructor.